core/pkg/sync/file/fileinfo_watcher.go

```diff
@@ -0,0 +1,204 @@
+package file
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/open-feature/flagd/core/pkg/logger"
+)
+
+// Implements file.Watcher using a timer and os.FileInfo
+type fileInfoWatcher struct {
+	// Event Chan
+	evChan chan fsnotify.Event
+	// Errors Chan
+	erChan chan error
+	// logger
+	logger *logger.Logger
+	// Func to wrap os.Stat (injection point for test helpers)
+	statFunc func(string) (fs.FileInfo, error)
```

Comment:
`statFunc` is nil outside of tests where you inject it.

To see this, If you do `make run-flagd` it runs flagd with a basic config and panic here.

Reply:
This is also why the integration test suite is failing as far as I can tell.

Reply:
You can run the e2e suite locally by basically running the 2 commands represented [here](https://github.com/open-feature/flagd/blob/239a432c18bf6780117b5d563443124887b38120/.github/workflows/build.yaml#L134-L144) (you'll have to update the paths to use `./` instead of the GH workspace.